src/generator/state-machine-to-intermediate.js

```diff
@@ -225,6 +244,36 @@ export default ({timers, states}) => {
         };
     });
 
+    // Generate numeric labels for each label name
+    states.forEach(({state, statements}, index) => {
+        // Generate numeric label if string
+        if(typeof state === 'string') {
+            if(!labelMap[state]) {
+                const start = generateNextLabel(states);
```

Comment:
Thinking about this more, I think that while it's intuitive to call `generateNextLabel` multiple times to create the symbol table, it makes the code more verbose than it needs to be and makes this step take O(n^2) time.

If we call the function once and store the result, and update it each time we process a labelled-state, we don't need to call the function again.

Reply:
I just commited an optimization for this that call endOfLastNumericState only one time